bindings.go

```diff
@@ -840,3 +757,48 @@ func (qc *QueryCursor) NextCapture() (*QueryMatch, uint32, bool) {
 
 	return qm, uint32(captureIndex), true
 }
+
+// keeps callbacks for parser.parse method
+type readFuncsMap struct {
+	sync.Mutex
+
+	funcs map[int]ReadFunc
+	count int
+}
+
+func (m *readFuncsMap) register(f ReadFunc) int {
+	m.Lock()
+	defer m.Unlock()
+
+	m.count++
+	m.funcs[m.count] = f
+	return m.count
+}
+
+func (m *readFuncsMap) unregister(id int) {
+	m.Lock()
+	defer m.Unlock()
+
+	delete(m.funcs, id)
+}
+
+func (m *readFuncsMap) get(id int) ReadFunc {
+	m.Lock()
+	defer m.Unlock()
+
+	return m.funcs[id]
+}
+
+//export callReadFunc
+func callReadFunc(id C.int, byteIndex C.uint32_t, position C.TSPoint, bytesRead *C.uint32_t) *C.char {
+	readFunc := readFuncs.get(int(id))
+	content := readFunc(uint32(byteIndex), Point{
+		Row:    uint32(position.row),
+		Column: uint32(position.column),
+	})
+	*bytesRead = C.uint32_t(len(content))
+
+	// Note: This memory is freed inside the C code; see above
```

Comment:
"see bindings.c", since the code has been moved.

Reply:
thanks! missed it.